### PR TITLE
Feature/auth

### PR DESCRIPTION
**Pull request by bgottlob**

### CODE REVIEW

File: apps/exile_auth/lib/exile_auth.ex [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  @doc """
+  Hello world.
+
+  ## Examples
+
+      iex> ExileAuth.hello()
+      :world
+
+  """
+  def hello do
```
Comment (holsee) [inline apps/exile_auth/lib/exile_auth.ex]: Unused code

File: apps/exile_auth/lib/exile_auth/accounts.ex [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+      iex> change_user(user)
+      %Ecto.Changeset{source: %User{}}
+
+  """
+  def change_user(%User{} = user) do
+    User.changeset(user, %{})
+  end
+
+  def authenticate_user(username, password) do
```
Comment (holsee) [inline apps/exile_auth/lib/exile_auth/accounts.ex:110]: needs doc

File: apps/exile_auth/test/exile_auth/accounts_test.exs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      user = user_fixture()
+      assert {:ok, %User{}} = Accounts.delete_user(user)
+      assert_raise Ecto.NoResultsError, fn -> Accounts.get_user!(user.id) end
+    end
+
+    test "change_user/1 returns a user changeset" do
+      user = user_fixture()
+      assert %Ecto.Changeset{} = Accounts.change_user(user)
+    end
+  end
```
Comment (holsee) [inline apps/exile_auth/test/exile_auth/accounts_test.exs:89]: auth user test missing?

**Comment by bgottlob** [comment]
I'm not using the two Plugs I created yet. Those will come into play when I start building the rest of the HTTP API.

File: apps/exile_auth/test/exile_auth_test.exs [hunk 4]
```diff
@@ -0,0 +1,8 @@
+defmodule ExileAuthTest do
+  use ExUnit.Case
+  doctest ExileAuth
+
+  test "greets the world" do
```
Comment (holsee) [inline apps/exile_auth/test/exile_auth_test.exs]: unused coded

File: apps/exile_web/lib/exile_web/controllers/auth_controller.ex [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+  alias ExileAuth.Accounts
+
+  def login(conn, %{"user" => %{"username" => username, "password" => password}}) do
+    Accounts.authenticate_user(username, password)
+    |> login_reply(conn)
+  end
+
+  defp login_reply({:ok, user}, conn) do
+    IO.puts(user.id)
```
Comment (holsee) [inline apps/exile_web/lib/exile_web/controllers/auth_controller.ex]: IO.puts left in

File: apps/exile_auth/lib/exile_auth/accounts.ex [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+    query = from u in User, where: u.username == ^username
+    case Repo.one(query) do
+      nil ->
+        Bcrypt.no_user_verify()
+        {:error, :invalid_credentials}
+      user ->
+        IO.puts(user.password)
```
Comment (holsee) [inline apps/exile_auth/lib/exile_auth/accounts.ex]: IO.puts left in

File: apps/exile_auth/lib/exile_auth/accounts.ex [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+        IO.puts(user.password_hash)
```
Comment (holsee) [inline apps/exile_auth/lib/exile_auth/accounts.ex]: IO.puts left in

File: apps/exile_auth/lib/exile_auth/guardian.ex [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+  end
+
+  def subject_for_token(_, _) do
+    {:error, :no_resource_id}
+  end
+
+  # Fetches the resource that is represented by claims.
+  # For JWT this would normally be found in the sub field.
+  def resource_from_claims(%{"sub" => id}) do
+    IO.puts(IO.inspect(id))
```
Comment (holsee) [inline apps/exile_auth/lib/exile_auth/guardian.ex]: IO.puts left in

File: config/config.exs [hunk 9]
```diff
@@ -17,6 +17,10 @@ config :exile_web,
   ecto_repos: [Exile.Repo],
   generators: [context_app: :exile]
 
+config :exile_auth,
+  ecto_repos: [Exile.Repo],
```
Comment (holsee) [inline config/config.exs]: A REPO should be owned by the OTP App using it, it should not be in the `exile` app

**Review by holsee — CHANGES_REQUESTED** [review]

**Pull request closed without merging**